src/java.base/windows/classes/sun/nio/ch/PipeImpl.java

```diff
@@ -205,8 +216,16 @@ public SinkChannelImpl sink() {
         return sink;
     }
 
-    private static ServerSocketChannel createListener() throws IOException {
-        ServerSocketChannel listener = ServerSocketChannel.open();
+    private static ServerSocketChannel createListener(boolean preferUnixDomain) throws IOException {
+        ServerSocketChannel listener;
+        if (preferUnixDomain && UnixDomainSockets.isSupported()) {
+            try {
+                listener = ServerSocketChannel.open(UNIX);
+                listener.bind(null);
+                return listener;
+            } catch (IOException | UnsupportedOperationException e) {}
```

Comment:
Is it possible isSupported to return true and open throw UOE?

Reply:
In practice no. I'll remove the UOE.

Reply:
Actually, it turns out this is not the entirely true. isSupported can return true, but if the temp directory (where unnamed sockets are created) is set to a non-default filesystem, then UOE does get thrown. We have a test that does this. I'll have to put it back.